### PR TITLE
Refactor3

### PR DESCRIPTION
**Pull request by SooLee**

### CODE REVIEW

**Comment by SooLee** [comment]
Basically the main changes are in `tibanna/awsem.py` (new), `tibanna_4dn/pony_utils.py` (new class `Fourfront_Updater`), and `tibanna_4dn/update_ffmeta.py` (simplified).

File: tibanna_4dn/pony_utils.py [hunk 1]
```diff
@@ -386,159 +395,761 @@ def is_prod():
...
+            pf_extra = self.pf_extra_file(pf_uuid, extra['file_format'])
+            extra_key = extra['upload_key']
+            # update the class object
+            pf_extra['md5sum'] = self.md5sum(pf_uuid=pf_uuid, secondary_key=extra_key)
+            pf_extra['file_size'] = self.filesize(pf_uuid=pf_uuid, secondary_key=extra_key)
+            pf_extra['status'] = 'uploaded'
+        # prepare for fourfront patch
+        self.add_to_pf_patch_items(pf_uuid, 'extra_files')
+
+    def add_higlass_to_pf(self, pf_uuid):
```
Comment (carlvitzthum) [inline tibanna_4dn/pony_utils.py]: Maybe`hgcf` is never set for files without `extra_files`? An example would be bigwig (bw) files

Comment (SooLee) [reply tibanna_4dn/pony_utils.py]: https://github.com/4dn-dcic/tibanna/blob/69892fb8477a6e0ffce85c5b216b06a7526b3675/tibanna_4dn/pony_utils.py#L883

Comment (carlvitzthum) [reply tibanna_4dn/pony_utils.py]: Okay, I see how it works. Was going a bit too fast when I made that comment!

File: tibanna_4dn/pony_utils.py [hunk 2]
```diff
@@ -386,159 +395,763 @@ def is_prod():
...
+                if hgcf:
+                    # register extra file not the original input file
+                    higlass_uid = register_to_higlass(self.tibanna_settings,
+                                                      self.bucket(ie.workflow_argument_name),
+                                                      self.file_key(ie.workflow_argument_name),
+                                                      hgcf['file_type'],
+                                                      hgcf['data_type'],
+                                                      ip.get('genome_assembly', None))
+                if higlass_uid:
+                    self.update_patch_item(ip['uuid'], {'higlass_uid': higlass_uid})
```
Comment (carlvitzthum) [inline tibanna_4dn/pony_utils.py]: Should this be `self.update_patch_items( ... )`?

Comment (SooLee) [reply tibanna_4dn/pony_utils.py]: Yes!! good catch!!

Comment (SooLee) [reply tibanna_4dn/pony_utils.py]: Fixed!

**Pull request merged**